src/sql/workbench/browser/scriptingUtils.ts

```diff
@@ -37,55 +37,61 @@ const targetDatabaseEngineEditionMap = {
-
-		return Boolean(innerConnectionResult) && innerConnectionResult.connected;
-	} else {
-		let errMsg: string = nls.localize('scriptSelectNotFound', "No script was returned when calling select script on object ");
-		throw new Error(errMsg.concat(metadata.metadataTypeName));
+export async function scriptSelect(connectionProfile: IConnectionProfile, metadata: azdata.ObjectMetadata, connectionService: IConnectionManagementService, queryEditorService: IQueryEditorService, scriptingService: IScriptingService, errorMessageService: IErrorMessageService): Promise<void> {
+	try {
+		const connectionResult = await connectionService.connectIfNotConnected(connectionProfile);
+		let paramDetails = getScriptingParamDetails(connectionService, connectionResult, metadata)!;
+		const result = await scriptingService.script(connectionResult, metadata, ScriptOperation.Select, paramDetails);
+		if (result && result.script) {
+			const owner = await queryEditorService.newSqlEditor({ initalContent: result.script }, connectionProfile?.providerName);
+			// Connect our editor to the input connection
+			let options: IConnectionCompletionOptions = {
+				params: { connectionType: ConnectionType.editor, runQueryOnCompletion: RunQueryOnConnectionMode.executeQuery, input: owner },
+				saveTheConnection: false,
+				showDashboard: false,
+				showConnectionDialogOnError: true,
+				showFirewallRuleOnError: true
+			};
+			await connectionService.connect(connectionProfile, owner.uri, options);
+		} else {
+			throw new Error(SelectScriptNotGeneratedError);
+		}
+	} catch (err) {
+		errorMessageService.showDialog(Severity.Error, ScriptingFailedDialogTitle, err?.message ?? err);
 	}
 }
 
 /**
  * Opens a new Edit Data session
  */
-export async function scriptEditSelect(connectionProfile: IConnectionProfile, metadata: azdata.ObjectMetadata, connectionService: IConnectionManagementService, queryEditorService: IQueryEditorService, scriptingService: IScriptingService): Promise<boolean> {
-	const connectionResult = await connectionService.connectIfNotConnected(connectionProfile);
-	let paramDetails = getScriptingParamDetails(connectionService, connectionResult, metadata);
-	const result = await scriptingService.script(connectionResult, metadata, ScriptOperation.Select, paramDetails!);
-	if (result && result.script) {
-		const owner = await queryEditorService.newEditDataEditor(metadata.schema, metadata.name, result.script);
-		// Connect our editor
-		let options: IConnectionCompletionOptions = {
-			params: { connectionType: ConnectionType.editor, runQueryOnCompletion: RunQueryOnConnectionMode.none, input: owner },
-			saveTheConnection: false,
-			showDashboard: false,
-			showConnectionDialogOnError: true,
-			showFirewallRuleOnError: true
-		};
-		const innerConnectionResult = await connectionService.connect(connectionProfile, owner.uri, options);
-
-		return Boolean(innerConnectionResult) && innerConnectionResult.connected;
```

Comment:
the return value from these functions wasn't actually being used by anything?

Reply:
these are only called by the command handlers, the return value is not used at all.